### NumPy Project: E-Commerce Sales Analysis

#### Problem Statement:

- An e-commerce company receives orders every day. Analyzing sales data manually takes time and may lead to mistakes. The company needs a simple way to analyze daily sales data and track performance.

#### Objectives:
- Store daily order data using a 1D NumPy array.
- Calculate total orders received.
- Find average daily orders.
- Identify highest and lowest sales days.
- Count days with above-average sales.
- Generate a simple sales performance report.

#### Solution:

In [1]:
import numpy as np

# Daily orders received in a week ( 7 days)
orders = np.array([120, 150, 180, 110, 220, 190, 140])     #1D Array

##### Sales Analysis:

In [2]:

total_orders=np.sum(orders)                  #Calculate total orders
average_orders=np.mean(orders)               #Calculate average orders

highest_order=np.max(orders)                  #Find the highest order
lowest_order=np.min(orders)                   #find the lowest order

sales_range = highest_order - lowest_order     #Difference between highest and lowest orders
median_orders = np.median(orders)              #middle order value after sorting the data

above_average_days=np.sum(orders>average_orders)   #Count days with sales above average


#### Performance Reporting:

In [3]:
print("Generate detailed insights about sales performance.\n")
print(" calculate total order : ",total_orders)
print(" calculate average orders : ",average_orders)
print(" calculate highest orders : ",highest_order)
print(" calculate Lowest orders : ",lowest_order)
print(" calculate total Sales range : ",sales_range)
print(" calculate median orders : ",median_orders)
print(" calculate average_days : ",above_average_days)

Generate detailed insights about sales performance.

 calculate total order :  1110
 calculate average orders :  158.57142857142858
 calculate highest orders :  220
 calculate Lowest orders :  110
 calculate total Sales range :  110
 calculate median orders :  150.0
 calculate average_days :  3


- A total of 1,110 orders were received during the week.
- The average number of orders is around 159 per day.
- The minimum orders are 110, and the maximum orders are 220.
- The sales range is 110, showing the difference between the highest and lowest sales days.

#### Business Impact
- Helps monitor overall sales performance.
- Identifies peak and low sales periods.
- Measures sales consistency and variation.
- Supports inventory and resource planning.
- Enables faster and more informed business decisions.
- Reduces manual effort in sales reporting.

### Conclusion

The E-Commerce Sales Analysis project demonstrates how NumPy can be used to analyze daily sales data efficiently. By calculating key performance metrics such as total orders, average orders, median orders, sales range, and above-average sales days, businesses can gain valuable insights into sales trends and improve decision-making.

### Final Project Summary

A NumPy-based sales analysis project that evaluates daily e-commerce orders using statistical and performance metrics. The project provides meaningful insights into sales behavior and demonstrates the practical application of NumPy for basic business analytics.